test/test_torch.py

```diff
@@ -6251,6 +6258,16 @@ def test_half_tensor(self):
             xh2 = torch.load(f)
             self.assertEqual(xh.float(), xh2.float())
 
+    def test_save_net_with_device(self):
+        net = NetwithDevice()
+        with tempfile.NamedTemporaryFile() as f:
+            torch.save(net, f)
+            f.seek(0)
+            net2 = torch.load(f)
+            self.assertEqual(type(net), type(net2))
+            self.assertEqual(net.state_dict(), net2.state_dict())
+            self.assertEqual(net.device, net2.device)
```

Comment:
Can we just check that we can serialize arbitrary `device` objects? The network is not necessary, because it doesn't change anything. Just create a list of devices that are different (`cpu`, `cpu:0`, `cuda`, `cuda:1`, etc.) and check that they are the same after pickling and loading.